### PR TITLE
Fix Dashboard Component

### LINKED ISSUE #8886: [UI] self.Meeshcard error on Dashboard
hey i am getting this error as `self.Meshcard is not a function`

[Slack Message](https://layer5io.slack.com/archives/CDM0ACDM5/p1695481670553669?thread_ts=1695481670.553669&cid=CDM0ACDM5)

### PR DESCRIPTION
**Pull request by aabidsofi19**
**Notes for Reviewers**

This PR fixes #8886

- I have reverted the conversion
- and reapplied @MUzairS15  cleanup 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by aabidsofi19** [comment]
@MUzairS15  can you confirm if i applied your changes properly

**Comment by saurabh100ni** [comment]
Let's discuss this on the Meshery dev call tomorrow,
Please add this as your agenda item in Docs: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit

**Comment by leecalcote** [comment]
Merge conflict, @aabidsofi19

File: ui/components/DashboardComponent.js [hunk 1]
```diff
@@ -120,248 +95,285 @@ const useStyles = makeStyles((theme) => ({
...
+    const self = this;
+    return getK8sClusterIdsFromCtxId(self.props?.selectedK8sContexts, self.props.k8sconfig)
+  }
 
-  const getK8sClusterIds = () => {
-    return getK8sClusterIdsFromCtxId(props.selectedK8sContexts, props.k8sconfig);
-  };
 
-  const setMeshScanData = (controlPlanesData, dataPlanesData) => {
+  fetchAvailableAdapters = () => {
```
Comment (MUzairS15) [inline ui/components/DashboardComponent.js:306]: Not Needed in Dashboard component.

File: ui/components/DashboardComponent.js [hunk 2]
```diff
@@ -370,107 +382,141 @@ const DashboardComponent=(props) => {
...
+    //   ),
+    // });
+    notify({ message : `${msg}`, event_type : EVENT_TYPES.ERROR })
+  };
+
+  handleDelete() {
+    return false;
+  }
+
+  handleAdapterClick = (adapterLoc) => () => {
```
Comment (MUzairS15) [inline ui/components/DashboardComponent.js:465]: This too.

File: ui/components/DashboardComponent.js [hunk 3]
```diff
@@ -480,310 +526,75 @@ const DashboardComponent=(props) => {
...
+        this.props.updateProgress({ showProgress : false });
+        if (typeof result !== "undefined") {
+          notify({ message : `${name} is connected at ${server}`, event_type : EVENT_TYPES.SUCCESS })
+        }
+      },
+      self.handleError("Could not connect to Kubernetes")
+    );
+  };
+
+  handleGrafanaClick = () => {
```
Comment (MUzairS15) [inline ui/components/DashboardComponent.js:538]: Remove this as well.

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

**Comment by aabidsofi19** [comment]
@theBeginner86  removed conflicts and applied uzairs suggestions

File: ui/components/Header.js [hunk 4]
```diff
@@ -586,6 +586,7 @@ class Header extends React.PureComponent {
       onDrawerCollapse,
       capabilityregistryObj,
     } = this.props;
+    console.log('rendering header');
```
Comment (theBeginner86) [inline ui/components/Header.js:589]:
```suggestion
```

Comment (aabidsofi19) [reply ui/components/Header.js]: just in case to see if it is still rerendering ..

**Pull request merged**